### ISSUE TITLE
VLC vs browser for unvetted audio/video files

### ISSUE THREAD

**Issue by joelewing** [drduh/macOS-Security-and-Privacy-Guide#404]
I should start by saying that I am by no means a security expert, just a curious bystander.

I was wondering if opening a mp4 or mp3 in Google Chrome is safer than opening it in VLC, given the rigorous sandbox that Chrome has.

Any thoughts?

**Comment by joelewing** [comment]
I agree with you, it seems preferable to pick a program that uses the App Sandbox. Apple Music and VLC do not. QuickTime does (though it looks like QuickTime is not "hardened"). But I was wondering if the sandboxing provided by Chrome was even more robust than what Apple offers.
<img width="302" alt="Screen Shot 2022-02-19 at 11 06 16 AM" src="https://user-images.githubusercontent.com/9755486/154811049-4095c6df-0321-44b1-83cd-66e95b125102.png">
<img width="302" alt="Screen Shot 2022-02-19 at 11 06 02 AM" src="https://user-images.githubusercontent.com/9755486/154811050-5f9f46cb-d517-4d99-beda-59a1f6b7ac1d.png">
<img width="304" alt="Screen Shot 2022-02-19 at 11 05 46 AM" src="https://user-images.githubusercontent.com/9755486/154811051-65af32ef-ffbd-49a9-9fe1-a539384616c0.png">

**Comment by beerisgood** [comment]
Was this test under a Mac with Intel CPU? Wonder how the result is with Apple M1. 

i also don’t know how this tool you used check for that stuff listed but maybe „hardening“ isn’t needed for sandboxed programs?

**Comment by joelewing** [comment]
This is from an Intel Mac, yes. The app is called Apparency. Hardening refers to turning off capabilities that the program does not require. [https://www.mothersruin.com/software/Apparency/use.html#code-info-signature](https://www.mothersruin.com/software/Apparency/use.html#code-info-signature)

It seems that the App Sandbox is more restrictive than hardening but doesn't cover all of the same capabilities. It's worth noting as well that apps from places outside of the App Store can still enable the App Sandbox. This was the case for the Apparency app itself.

**Comment by drduh** [comment]
I can say from experience this is true for PDF files: it is safer to open them in Chrome where sandboxing is stronger than in native or third party apps. It is probably quite good at sandboxing other file types as well, but we should hear from an expert and make a recommendation accordingly. Thanks for bringing this up.

**Comment by reactive-firewall** [comment]
🤷🏻 Caveat: I should start by disclaiming any expertise on the security details, including for mp4 files. Let's say I'm also a curious bystander...

🤔 I see a few parts to this question that can lead to different answers:
Apple likes to use their Core-Audio throughout macOS and opportunistically (can I say aggressively) seem to convert media to ac3 (in an mpeg v4 container) core audio for their hardware (namely Safari and the ShareKit features alter audio to mp4) so the browser may not be the only issue rather both the browser handling and subsequent handoff to the audio engine should not be one and the same.

💭 For the browser handling side, if you are concerned about unvetted mp4 files I'd recommend the usual secure environment (hardened VM or the like) file download (for example `curl` straight to a file - guru's to weigh in on the details) from there `ffmpeg` has a probe tool for getting to the bottom of how the mp4 file is actually crafted called 'ffprobe'

❔ for the transfer to the core-audio library I'd actually welcome any thoughts and knowhow on the security considerations of MacOS's core-audio components. Especially in regards to any secure configuration of them.

**Comment by drduh** [comment]
We could benefit from additional expertise on this topic, or add general information on additional sandboxing/virtualization techniques for opening some file types.

**Comment by kimg45** [comment]
So looked into it a bit and there's some considerations here:

1. VLC requires you to install Rosetta which adds extra attack surface and performance overhead, they seem keen on backwards compatibility with older versions of macOS so I don't see this changing anytime soon.
2. I don't really want to trust third party tools for information like this so I found out that you can see which apps are sandboxed in Activity Monitor by right clicking the columns and checking "Sandbox". There's also "Restricted" but I couldn't find any info on what that means, I assume it's good tho. You can see that Chrome doesn't use the App Sandbox, although they have their own strong sandbox so I'm sure it's not an issue.
3. If you run the command:
`codesign --display --verbose /System/Applications/Music.app`
You can see what the developers set under "flags". So if you see (runtime) then they enabled the hardened runtime (this is a requirement for [notarization](https://developer.apple.com/documentation/security/notarizing_macos_software_before_distribution)) which VLC does which is good. Although you can see that the runtime version for VLC is much lower than any of the other apps in question which I'm sure comes with security implications. If you check QuickTime Player, it shows that it in fact does have Hardened Runtime enabled so it seems the third party tool you were using gave you bad info. As far as notarization goes, of course the official Apple apps don't need it since they're already reviewed by Apple in house.

So, in conclusion, 1. don't trust third party apps to tell you accurate info on your system and 2. can't say which of these options is objectively the best but VLC is pretty much the worst in terms of security imo

**Comment by beerisgood** [comment]
Point 3 & 4:
a much easier solution for the user is using the Activity monitor and checking (after adding the column) the Sandbox/ Restricted status.

https://developer.apple.com/documentation/security/app_sandbox/protecting_user_data_with_app_sandbox#4098972
![](https://docs-assets.developer.apple.com/published/67f023afa2/renderedDark2x-1667920132.png)

**Comment by kimg45** [comment]
> Point 3 & 4: a much easier solution for the user is using the Activity monitor and checking (after adding the column) the Sandbox/ Restricted status.
> 
> https://developer.apple.com/documentation/security/app_sandbox/protecting_user_data_with_app_sandbox#4098972 ![](https://camo.githubusercontent.com/2d68ad5334648408491f9ba406726004299c25c80ae425c653ec4bfd439bf1f4/68747470733a2f2f646f63732d6173736574732e646576656c6f7065722e6170706c652e636f6d2f7075626c69736865642f363766303233616661322f72656e64657265644461726b32782d313636373932303133322e706e67)

Yes, unfortunately it doesn't show whether the Hardened Runtime is enabled. You could just try to run it and see if Gatekeeper blocks you I suppose, that's probably the easiest way.

**Comment by beerisgood** [comment]
> > Point 3 & 4: a much easier solution for the user is using the Activity monitor and checking (after adding the column) the Sandbox/ Restricted status.
> > https://developer.apple.com/documentation/security/app_sandbox/protecting_user_data_with_app_sandbox#4098972 ![](https://camo.githubusercontent.com/2d68ad5334648408491f9ba406726004299c25c80ae425c653ec4bfd439bf1f4/68747470733a2f2f646f63732d6173736574732e646576656c6f7065722e6170706c652e636f6d2f7075626c69736865642f363766303233616661322f72656e64657265644461726b32782d313636373932303133322e706e67)
> 
> Yes, unfortunately it doesn't show whether the Hardened Runtime is enabled. You could just try to run it and see if Gatekeeper blocks you I suppose since all notarized apps are required to enable it, that's probably the easiest way.

Are you sure? For me that sounds like „Sandbox“ in activity monitor is Hardened Runtime.

**Comment by kimg45** [comment]
From the [documentation](https://developer.apple.com/documentation/security/hardened_runtime):

>The Hardened Runtime, along with System Integrity Protection (SIP), protects the runtime integrity of your software by preventing certain classes of exploits, like code injection, dynamically linked library (DLL) hijacking, and process memory space tampering. To enable the Hardened Runtime for your app, navigate in Xcode to your target’s Signing & Capabilities information and click the + button. In the window that appears, choose Hardened Runtime.

And the App [Sandbox](https://developer.apple.com/documentation/security/app_sandbox/protecting_user_data_with_app_sandbox):

>The operating system creates a container directory when launching your sandboxed app, to which the app has unrestricted read and write access. The sandboxed app doesn’t have unrestricted access to the user’s home folder.

So basically the Hardened runtime disables stuff like Just-In-Time compilation that has to do with the app itself, and the App Sandbox restricts its access to other files and things on your system.

**Comment by beerisgood** [comment]
Alright. Thanks for the Correction.
